tests/test_envs.py

```diff
@@ -58,7 +58,7 @@ async def my_policy(obs: list[Message]) -> ToolRequestMessage:  # noqa: ARG001,
             action=ToolRequestMessage(tool_calls=[])
         )
         assert not done, "Should not be done after empty action"
-        assert "no tool calls" in obs[0].content.lower()
+        assert "no tool calls" in obs[0].content.lower()  # type: ignore[union-attr]
```

Comment:
How to properly fix this is to do two assertions, one confirming `str` in `str | None`, then second confirms lower

```python
        assert obs[0].content
        assert "no tool calls" in obs[0].content.lower()
```